### PR TITLE
Added new operation: Get vm overrides

### PR DESCRIPTION
**Pull request by Tirla-Alin**
Added new operation: Edit vm overrides
Added unit tests for the operation
Added operation description

Signed-off-by: Tirla Florin-Alin <tirla@hpe.com>

### CODE REVIEW

File: cs-vmware/src/main/java/io/cloudslang/content/vmware/actions/cluster/GetVmOverrides.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+     *                              Default Value: "false"
+     * @param hostname              The name of the target host to be queried to retrieve the supported guest OSes
+     *                              Example: "host123.subdomain.example.com"
+     * @param virtualMachineName    The name of the virtual machine for which the override will be created. This input
+     *                              is mutually exclusive with virtualMachineId.
+     * @param virtualMachineId      The id of the virtual machine for which the override will be created. This input is
+     *                              mutually exclusive with virtualMachineName.
+     *                              Example: "vm-1230"
+     * @param clusterName           The name of the cluster.
+
```
Comment (victorursan) [inline cs-vmware/src/main/java/io/cloudslang/content/vmware/actions/cluster/GetVmOverrides.java]: where is the * ?

File: cs-vmware/src/main/java/io/cloudslang/content/vmware/services/ClusterComputeResourceService.java [hunk 2]
```diff
@@ -89,6 +94,27 @@
...
+    public String getVmOverride(final HttpInputs httpInputs, final VmInputs vmInputs) throws Exception {
+        final ConnectionResources connectionResources = new ConnectionResources(httpInputs, vmInputs);
+
+        final ManagedObjectReference clusterMor = new MorObjectHandler().getSpecificMor(connectionResources, connectionResources.getMorRootFolder(),
+                ClusterParameter.CLUSTER_COMPUTE_RESOURCE.getValue(), vmInputs.getClusterName());
+
+        final ClusterConfigInfoEx clusterConfigInfoEx = getClusterConfiguration(connectionResources, clusterMor, vmInputs.getClusterName());
+
+        final String restartPriority;
+        if(isNotBlank(vmInputs.getVirtualMachineId()) || isNotBlank(vmInputs.getVirtualMachineName())) {
```
Comment (victorursan) [inline cs-vmware/src/main/java/io/cloudslang/content/vmware/services/ClusterComputeResourceService.java]: you either don't import 'isNotBlank' statically or import 'StringUtilities.isNotEmpty' (and other StringUtilities methods used in the class) statically, we should keep the class consistent

File: cs-vmware/src/main/java/io/cloudslang/content/vmware/utils/InputUtils.java [hunk 3]
```diff
@@ -138,4 +139,10 @@ public static void checkMutuallyExclusiveInputs(final String input1, final Strin
             throw new IllegalArgumentException(exceptionMessage);
         }
     }
+
+    public static void checkOptionalMutuallyExclusiveInputs(final String input1, final String input2, final String exceptionMessage) {
+        if (isNotBlank(input1) && isNotBlank(input2)) {
```
Comment (victorursan) [inline cs-vmware/src/main/java/io/cloudslang/content/vmware/utils/InputUtils.java]: does this check mutually exclusivity? the method name is confusing 
if yes, this doesn't check if both are empty

Comment (Tirla-Alin) [inline cs-vmware/src/main/java/io/cloudslang/content/vmware/utils/InputUtils.java]: This is a method that checks optionally mutual exclusive (you can specify one of them or none of them; not providing value for any of the inputs is a valid use-case).

File: cs-vmware/src/main/java/io/cloudslang/content/vmware/actions/cluster/GetVmOverrides.java (hunk 1 above)
Comment (Tirla-Alin) [inline cs-vmware/src/main/java/io/cloudslang/content/vmware/actions/cluster/GetVmOverrides.java]: My mistake, should not have that empty line at all.

File: cs-vmware/src/main/java/io/cloudslang/content/vmware/services/ClusterComputeResourceService.java (hunk 2 above)
Comment (Tirla-Alin) [inline cs-vmware/src/main/java/io/cloudslang/content/vmware/services/ClusterComputeResourceService.java]: Ok.

File: cs-vmware/src/main/java/io/cloudslang/content/vmware/utils/InputUtils.java (hunk 3 above)
Comment (victorursan) [inline cs-vmware/src/main/java/io/cloudslang/content/vmware/utils/InputUtils.java]: Ob, but then the description of the inputs is not clear, 
```
     * @param virtualMachineName    The name of the virtual machine for which the override will be created. This input
     *                              is mutually exclusive with virtualMachineId.
     * @param virtualMachineId      The id of the virtual machine for which the override will be created. This input is
     *                              mutually exclusive with virtualMachineName.
```
at least add that they are optional

Comment (Tirla-Alin) [inline cs-vmware/src/main/java/io/cloudslang/content/vmware/utils/InputUtils.java]: Done

**Pull request merged**